lib/saml_authenticator.rb

```diff
@@ -156,19 +156,20 @@ def after_authenticate(auth)
 
     if result.user.blank?
       result.username = '' if setting(:clear_username)
-      result.omit_username = true if setting(:omit_username)
       result.user = auto_create_account(result, uid) if setting(:auto_create_account) && result.email_valid
     else
       @user = result.user
       sync_groups
       sync_custom_fields
-      sync_email(result.email, uid)
       sync_moderator
       sync_admin
       sync_trust_level
       sync_locale
     end
 
+    result.overrides_username = setting(:omit_username)
+    result.overrides_email = setting(:sync_email)
+
```

Comment:
This `after_authenticate` method is huge. Don't wanna block you, but I feel this result addition should have some side effect in some test.

Reply:
Fair point. I'll see about adding an integration test for this functionality

Reply:
Integration test added (although note that it requires https://github.com/discourse/discourse/pull/15378, which in turn requires https://github.com/discourse/discourse-oauth2-basic/pull/56,  before it'll pass)